swift/Tests/Blockchains/OntologyTests.swift

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import TrustWalletCore
+import XCTest
+
+extension String {
+    public func indexOf<T>(stuff: T) -> Int where T: StringProtocol {
+        let r = range(of: stuff)
+        guard let rr = r else {
+            return -1
+        }
+        return rr.lowerBound.encodedOffset
+    }
+}
+
+class OntologyTests: XCTestCase {
+    
+    func testSignOngBalanceOf() {
+        
+        let input = TWOntologySigningInput.with {
+            $0.contract = "ONG".data(using: .utf8)!
+            $0.method = "balanceOf".data(using: .utf8)!
+            $0.queryAddress = "ANDfjwrUroaVtvBguDtrWKRMyxFwvVwnZD".data(using: .utf8)!
+        }
+        
+        let output = OntologySigner.sign(input: input)
```

Comment:
This is just encoding? no private key involved

Reply:
The `balanceOf` transaction will be send into network in pre-execution mode, which will not consume any gas, the transaciton will alson not be record into block. This type of transaction only to run in VM and get the running result. I think your opinion is correct, I want to correct it's name. I want to know whether can I change `OntologySigner` to other more suitable name.

Reply:
I think you can add a `encode` method to the signer

Reply:
I had change it to `encode`, which is more suitable than `sign`. thanks for your help. https://github.com/TrustWallet/wallet-core/pull/201/commits/945584af3faee61f06aa5b8a0652e3616c7ce222